rust/src/mqtt/mqtt.rs

```diff
@@ -749,7 +750,7 @@ export_state_data_get!(rs_mqtt_get_state_data, MQTTState);
 #[no_mangle]
 pub unsafe extern "C" fn rs_mqtt_register_parser(cfg_max_msg_len: u32) {
     let default_port = CString::new("[1883]").unwrap();
-    let max_msg_len = &mut MAX_MSG_LEN;
+    let max_msg_len = addr_of_mut!(MAX_MSG_LEN);
```

Comment:
this is the first time I see this in use for us, why do we need it? master also doesn't have it

Reply:
Could we just do:
```diff
diff --git a/rust/src/mqtt/mqtt.rs b/rust/src/mqtt/mqtt.rs
index 8260251218..35f3c6d459 100644
--- a/rust/src/mqtt/mqtt.rs
+++ b/rust/src/mqtt/mqtt.rs
@@ -749,8 +749,7 @@ export_state_data_get!(rs_mqtt_get_state_data, MQTTState);
 #[no_mangle]
 pub unsafe extern "C" fn rs_mqtt_register_parser(cfg_max_msg_len: u32) {
     let default_port = CString::new("[1883]").unwrap();
-    let max_msg_len = &mut MAX_MSG_LEN;
-    *max_msg_len = cfg_max_msg_len;
+    MAX_MSG_LEN = cfg_max_msg_len;
     let parser = RustParser {
         name: PARSER_NAME.as_ptr() as *const std::os::raw::c_char,
         default_port: default_port.as_ptr(),
```

Reply:
This was "accidentally" fixed in master by f12e0266967a526e0d7ad0ef9cb8fecb126f3889

Thanks Jason, your fix is better than the one proposed by the compiler